js_modules/dagster-ui/packages/ui-core/src/assets/insights/AssetCatalogHorizontalTopAssetsChart.tsx

```diff
@@ -1,148 +1,266 @@
 import {
-  Body,
-  BodySmall,
   Box,
-  Button,
   Colors,
+  Container,
   Icon,
+  Inner,
+  ListItem,
   MiddleTruncate,
   Mono,
-  Spinner,
+  Row,
+  UnstyledButton,
 } from '@dagster-io/ui-components';
-import React, {useMemo, useState} from 'react';
+import {useVirtualizer} from '@tanstack/react-virtual';
+import clsx from 'clsx';
+import React, {useMemo, useRef, useState} from 'react';
 import {Link} from 'react-router-dom';
 
 import styles from './AssetCatalogHorizontalTopAssetsChart.module.css';
+import {COMMON_COLLATOR} from '../../app/commonCollator';
 import {tokenToAssetKey} from '../../asset-graph/Utils';
+import {formatMetric} from '../../insights/formatMetric';
+import {ReportingUnitType} from '../../insights/types';
 import {numberFormatter} from '../../ui/formatters';
 import {assetDetailsPathForKey} from '../assetDetailsPathForKey';
 
-const PAGE_SIZE = 10;
+type SortBy = 'name' | 'current' | 'previous' | 'change';
+type SortDir = 'asc' | 'desc';
 
-const DistributionChartRow = ({maxValue, value}: {maxValue: number; value: number}) => {
-  const percentage = Math.round((value / maxValue) * 100);
+const getInitialSortBy = (currentData: MaybeData, previousData: MaybeData) => {
+  return currentData && previousData
+    ? 'change'
+    : currentData
+      ? 'current'
+      : previousData
+        ? 'previous'
+        : 'name';
```

Comment:
I'd prefer not using ternaries this deep 😬

Reply:
Yeah, these are unpleasant. I'll fix them up.